# COGS 108 - Final Project 

## Permissions

Place an `X` in the appropriate bracket below to specify if you would like your group's project to be made available to the public. (Note that PIDs will be scraped from the public submission, but student names will be included.)

* [  ] YES - make available
* [ x ] NO - keep private

# Overview

There have been some debates about the best wine overall and favorites of top 19 wine enthusiasts, or which country relatively produces better wines compared to the others, for example: U.S vs European countries. We would take statistical analysis on the wine review dataset to identify the highest rated wines, while comparing between wineries, vineyards, type of grapes, prices, and regions. We would then reach the conclusion regarding which wine is the best wine. 

# Names

- Devin Junanto
- Gabriela Shirley
- Natalia Sudjana
- Shih Gau Peng

# Group Members IDs

- A15754079
- A16095578
- A16112711
- A16059733

# Research Question

1. What is the best wine overall based on the ratings given by wine enthusiasts?
2. Which continent, and country more specifically, produce the most high-rated wines? 
3. What type of grapes relatively produce the most high-rated wines?
4. Does the age (in year) of the wine as well as the price determine the wine ratings?
5. Which wines are good recommendations to the wine tasters based on their ‘winery’ and ‘types of grapes’ preferences?
6. Predict the points given to a new wine based on its winery, variety (type of grapes), country, continent, age, and price.

## Background and Prior Work

Winer are ever changing in relation to time. People’s taste buds vary but there may be a similarity of tastes once we analyze our big data sets. Every county, province, state, and country offers a breadth of wine varietal. Thus, through this project, we put high interest to identify which region produces the high-rated wines based on the points given by nineteen wine enthusiasts.

Based on some prior works that we looked into, the best wine overall is Promontory 2013 from Napa Valley, U.S.2. While from another source, Chateau Montelena Napa Valley Chardonnay 2016 from Napa Valley, U.S.3 is on the top of the list, with another similar wine type named Chateau Leoville Barton St. Julien 2016 from France that also placed first out of 100 on a different source4. By comparing these top three sources from the internet, we could see how different their answers to the question “What is the best wine overall?”. Looking at varying sources, we could not rely on one as each source would have different research techniques to build up their lists. Therefore, as to fulfill our team’s curiosity of the real wine ratings and countries that produce them, we would analyze the “Wine Reviews” dataset1 from 2018 to come up with answers to our research questions.

The wine datasets from Kaggle gathered as much as 130,000 wines which contain several variables such as the variety, winery, price, description, and ratings as well as where they were produced which would help us analyze the findings and recommend wines to the wine tasters . We would like to gather what we have learned in class and use that knowledge to wrangle and analyze data until we come to a conclusion, while we are taking into account the people's privacy.

References (include links):

1) https://www.kaggle.com/zynicide/wine-reviews#winemag-data_first150k.csv

2) https://robbreport.com/best-of-the-best/2019-best-wine-285174

3) https://vinepair.com/articles/best-wines-2019/

4) https://top100.winespectator.com/lists/


# Hypothesis


1. We hypothesize that the type of wines that wine testers prefer come from they type of grape called Chardonnay as the type of grapes most preferred as well as the highest rated wine being the Chateau Montelena Napa Valley Chardonnay 2016. 
2. We hypothesize as the wine ages more, they increase in value in terms of quality and price which can increase the ratings of the wine as more wine enthusiasts enjoy an aged wine.
3. We hypothesize that Europe as a continent and Russia as a country produces the highest rated wines as they are well known for their wines as well as Russia being the most populated country in Europe.

# Dataset(s)

Dataset Name: Wine Reviews 2018

Link to the dataset: https://www.kaggle.com/zynicide/wine-reviews#winemag-data-130k-v2.csv

Number of observations: 130,000 rows, 14 columns

Description: this dataset was scraped from WineEnthusiast around June 2017, and updated around November 2017 which contains 130,000 wine reviews with variety, location, winery, price, and descriptions.

# Setup

##### 1.1 Import library

In [1]:
import numpy as np
import gzip
import urllib.request
import tarfile
import random
import string
from copy import deepcopy
import pandas as pd
from collections import defaultdict
import json
from sklearn.metrics import mean_squared_error
from sklearn import linear_model
from sklearn.svm import LinearSVC
import matplotlib.pyplot as plt
import re
import matplotlib as mpl
import matplotlib.pyplot as plt
import patsy
import statsmodels.api as sm
import seaborn as sns
%matplotlib notebook

In [2]:
##### 1.2 Read data from json file

In [3]:
df_wine = pd.read_json("winemag-data-130k-v2.json")

In [4]:
##### 1.3 Check column labels

In [5]:
df_wine.columns.values

array(['points', 'title', 'description', 'taster_name',
       'taster_twitter_handle', 'price', 'designation', 'variety',
       'region_1', 'region_2', 'province', 'country', 'winery'],
      dtype=object)

# Data Cleaning

##### 2.1 Remove unnecessary columns

In [6]:
df_wine = df_wine.drop(['province', 'region_1', 'region_2', 'description', 'taster_twitter_handle'], axis=1)

In [7]:
df_wine

points                                              title  \
0           87                  Nicosia 2013 Vulkà Bianco  (Etna)   
1           87      Quinta dos Avidagos 2011 Avidagos Red (Douro)   
2           87      Rainstorm 2013 Pinot Gris (Willamette Valley)   
3           87  St. Julian 2013 Reserve Late Harvest Riesling ...   
4           87  Sweet Cheeks 2012 Vintner's Reserve Wild Child...   
...        ...                                                ...   
129966      90  Dr. H. Thanisch (Erben Müller-Burggraef) 2013 ...   
129967      90                  Citation 2004 Pinot Noir (Oregon)   
129968      90  Domaine Gresser 2013 Kritt Gewurztraminer (Als...   
129969      90      Domaine Marcel Deiss 2012 Pinot Gris (Alsace)   
129970      90  Domaine Schoffit 2012 Lieu-dit Harth Cuvée Car...   

               taster_name  price                             designation  \
0            Kerin O’Keefe    NaN                            Vulkà Bianco   
1               Roger Voss   15.0                                Avidagos   
2             Paul Gregutt   14.0                                    None   
3       Alexander Peartree   13.0                    Reserve Late Harvest   
4             Paul Gregutt   65.0      Vintner's Reserve Wild Child Block   
...                    ...    ...                                     ...   
129966  Anna Lee C. Iijima   28.0  Brauneberger Juffer-Sonnenuhr Spätlese   
129967        Paul Gregutt   75.0                                    None   
129968          Roger Voss   30.0                                   Kritt   
129969          Roger Voss   32.0                                    None   
129970          Roger Voss   21.0           Lieu-dit Harth Cuvée Caroline   

               variety   country                                    winery  
0          White Blend     Italy                                   Nicosia  
1       Portuguese Red  Portugal                       Quinta dos Avidagos  
2           Pinot Gris        US                                 Rainstorm  
3             Riesling        US                                St. Julian  
4           Pinot Noir        US                              Sweet Cheeks  
...                ...       ...                                       ...  
129966        Riesling   Germany  Dr. H. Thanisch (Erben Müller-Burggraef)  
129967      Pinot Noir        US                                  Citation  
129968  Gewürztraminer    France                           Domaine Gresser  
129969      Pinot Gris    France                      Domaine Marcel Deiss  
129970  Gewürztraminer    France                          Domaine Schoffit  

[129971 rows x 8 columns]

##### 2.2 Dropping all the none values

In [8]:
df_wine = df_wine[df_wine.country != 'nan']
df_wine = df_wine[df_wine.points != 'nan']
df_wine = df_wine[df_wine.price != 'nan']
df_wine = df_wine[df_wine.title != 'nan']
df_wine = df_wine[df_wine.variety != 'nan']
df_wine = df_wine[df_wine.winery != 'nan']

/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  res_values = method(rvalues)


##### 2.3 Get the continent based on the country given

In [9]:
def continent(country):
    if country in ['US', 'Canada', 'Mexico']:
        return 'North America'
    elif country in ['France', 'Italy', 'Spain', 'Portugal', 'Austria', 'Germany', 'Greece', 'Hungary', 'Bulgaria', 'Romania'
                     , 'Turkey', 'Slovenia', 'Georgia', 'England', 'Croatia', 'Moldova', 'Ukraine', 'Macedonia', 'Czech Republic'
                     , 'Serbia', 'Cyprus', 'Switzerland', 'Luxembourg', 'Bosnia and Herzegovina', 'Armenia', 'Slovakia']:
        return 'Europe'
    elif country in ['Chile', 'Argentina', 'Uruguay', 'Brazil', 'Peru']:
        return 'South America'
    elif country in ['Israel', 'India', 'China']:
        return 'Asia'
    elif country in ['Australia', 'New Zealand']:
        return 'Oceania'
    elif country in ['South Africa', 'Egypt', 'Morocco']:
        return 'Africa'
    else:
        return np.nan

In [10]:
df_wine['continent'] = df_wine['country'].apply(continent)

##### 2.4 Get the year and age from the title of the wine

In [11]:
def getYear(string):
    splitStr = []
    splitStr = string.split(' ')
    for a in splitStr:
        if a.isnumeric() == True:
            if int(a) >= 1934 and int(a) <= 2020:
                return a
            else:
                continue
        else:
            continue
    return np.nan

In [12]:
df_wine['year'] = df_wine['title'].apply(getYear)

In [13]:
def getAge(year):
    if type(year) == float:
        return np.nan
    else:
        age = 2018 - int(year)
        return age

In [14]:
df_wine['age'] = df_wine['year'].apply(getAge)

##### 2.5 Get the description of the data

In [15]:
df_wine.head()

points                                              title  \
0      87                  Nicosia 2013 Vulkà Bianco  (Etna)   
1      87      Quinta dos Avidagos 2011 Avidagos Red (Douro)   
2      87      Rainstorm 2013 Pinot Gris (Willamette Valley)   
3      87  St. Julian 2013 Reserve Late Harvest Riesling ...   
4      87  Sweet Cheeks 2012 Vintner's Reserve Wild Child...   

          taster_name  price                         designation  \
0       Kerin O’Keefe    NaN                        Vulkà Bianco   
1          Roger Voss   15.0                            Avidagos   
2        Paul Gregutt   14.0                                None   
3  Alexander Peartree   13.0                Reserve Late Harvest   
4        Paul Gregutt   65.0  Vintner's Reserve Wild Child Block   

          variety   country               winery      continent  year  age  
0     White Blend     Italy              Nicosia         Europe  2013  5.0  
1  Portuguese Red  Portugal  Quinta dos Avidagos         Europe  2011  7.0  
2      Pinot Gris        US            Rainstorm  North America  2013  5.0  
3        Riesling        US           St. Julian  North America  2013  5.0  
4      Pinot Noir        US         Sweet Cheeks  North America  2012  6.0

# Data Analysis & Results

##### 3.1 Get the average of wine ratings

In [16]:
df_wine['points'].mean()

88.44713820775404

##### 3.2 Get the number of wine tasters

In [17]:
df_wine['taster_name'].value_counts().count()

19

##### 3.3 Get the number of the wine title

In [18]:
df_wine['title'].value_counts().count()

118840

##### 3.4 Get the number of the winery

In [19]:
df_wine['winery'].value_counts().count()

16757

##### 3.5 Get the number of the country

In [20]:
df_wine['country'].value_counts().count()

43

##### 3.6 Get the average rating that each wine taster gives to get a sense of their overall standard

In [21]:
taster = []
taster = df_wine['taster_name'].value_counts().keys().to_list()
average_taster = []
for a in taster:
    average_taster.append(df_wine[df_wine['taster_name'] == a].points.mean())
df_taster = pd.DataFrame({'taster_name':taster, 'points':average_taster})
df_taster = df_taster.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_taster = df_taster.set_index('taster_name')

In [22]:
df_taster

points
taster_name                  
Anne Krebiehl MW    90.562551
Matt Kettmann       90.008686
Virginie Boone      89.213379
Mike DeSimone       89.101167
Paul Gregutt        89.082564
Kerin O’Keefe       88.867947
Sean P. Sullivan    88.755739
Roger Voss          88.708003
Jim Gordon          88.626287
Joe Czerwinski      88.536235
Anna Lee C. Iijima  88.415629
Jeff Jenssen        88.319756
Christina Pickard   87.833333
Lauren Buzzeo       87.739510
Michael Schachner   86.907493
Fiona Adams         86.888889
Susan Kostrzewa     86.609217
Carrie Dykes        86.395683
Alexander Peartree  85.855422

##### 3.7 Get the continents that produce the best wines based on the wine tasters' ratings

In [23]:
continent = []
continent = df_wine['continent'].value_counts().keys().to_list()
average_continent = []
for a in continent:
    average_continent.append(df_wine[df_wine['continent'] == a].points.mean())
df_continent = pd.DataFrame({'continent':continent, 'points':average_continent})
df_continent = df_continent.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_continent = df_continent.set_index('continent')

In [24]:
df_continent

points
continent               
Europe         88.608982
North America  88.563276
Asia           88.502913
Oceania        88.475454
Africa         88.063636
South America  86.577583

##### 3.8 Get the countries that produce the best wines based on the wine tasters' ratings

In [25]:
country = []
country = df_wine['country'].value_counts().keys().to_list()
average_country = []
for a in country:
    average_country.append(df_wine[df_wine['country'] == a].points.mean())
df_country = pd.DataFrame({'country':country, 'points':average_country})
df_country = df_country.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_country = df_country.set_index('country')

In [26]:
df_country

points
country                          
England                 91.581081
India                   90.222222
Austria                 90.101345
Germany                 89.851732
Canada                  89.369650
Hungary                 89.191781
China                   89.000000
France                  88.845109
Luxembourg              88.666667
Australia               88.580507
Switzerland             88.571429
Morocco                 88.571429
US                      88.563720
Italy                   88.562231
Israel                  88.471287
New Zealand             88.303030
Portugal                88.250220
Turkey                  88.088889
Slovenia                88.068966
South Africa            88.056388
Bulgaria                87.936170
Georgia                 87.686047
Lebanon                 87.685714
Armenia                 87.500000
Serbia                  87.500000
Spain                   87.288337
Greece                  87.283262
Czech Republic          87.250000
Croatia                 87.219178
Moldova                 87.203390
Cyprus                  87.181818
Slovakia                87.000000
Macedonia               86.833333
Uruguay                 86.752294
Argentina               86.710263
Bosnia and Herzegovina  86.500000
Chile                   86.493515
Romania                 86.400000
Mexico                  85.257143
Brazil                  84.673077
Ukraine                 84.071429
Egypt                   84.000000
Peru                    83.562500

##### 3.9 Get the wineries that produce the best wines based on the wine tasters' ratings

In [27]:
winery = df_wine['winery'].unique()
winery

array(['Nicosia', 'Quinta dos Avidagos', 'Rainstorm', ...,
       'Mas de Pampelonne', 'Bodegas Eidosela', 'Penedo Borges'],
      dtype=object)

In [28]:
avg1 = []
for i in winery:
    avg1.append(df_wine[df_wine.winery == i]['points'].mean())

In [29]:
wavg_points = {'Winery': winery, 'Average Points': avg1} 

wavg_points = pd.DataFrame(wavg_points) 
 
wavg_points = wavg_points.sort_values(by=['Average Points'], ascending=False)

wavg_points

Winery  Average Points
11466              Araujo            98.0
16486             Gandona            97.0
16485                Ovid            97.0
15600          J.L. Chave            97.0
7914                Salon            96.8
...                   ...             ...
16310        Gran Familia            80.0
7130           Villa Viña            80.0
16334        Capay Valley            80.0
14868           Moss Roxx            80.0
8075   California's Jewel            80.0

[16757 rows x 2 columns]

##### 3.10 Get the best type of grapes based on the wine tasters' ratings

In [30]:
grapes = df_wine['variety'].unique()
grapes

array(['White Blend', 'Portuguese Red', 'Pinot Gris', 'Riesling',
       'Pinot Noir', 'Tempranillo-Merlot', 'Frappato', 'Gewürztraminer',
       'Cabernet Sauvignon', 'Nerello Mascalese', 'Chardonnay', 'Malbec',
       'Tempranillo Blend', 'Meritage', 'Red Blend', 'Merlot',
       "Nero d'Avola", 'Chenin Blanc', 'Gamay', 'Sauvignon Blanc',
       'Viognier-Chardonnay', 'Primitivo', 'Catarratto', 'Inzolia',
       'Petit Verdot', 'Monica', 'Bordeaux-style White Blend', 'Grillo',
       'Sangiovese', 'Cabernet Franc', 'Champagne Blend',
       'Bordeaux-style Red Blend', 'Aglianico', 'Petite Sirah',
       'Touriga Nacional', 'Carmenère', 'Albariño', 'Petit Manseng',
       'Rosé', 'Zinfandel', 'Vernaccia', 'Rosato', 'Grüner Veltliner',
       'Viognier', 'Vermentino', 'Grenache Blanc', 'Syrah', 'Nebbiolo',
       'Shiraz-Cabernet Sauvignon', 'Pinot Blanc', 'Alsace white blend',
       'Barbera', 'Rhône-style Red Blend', 'Portuguese White', 'Graciano',
       'Tannat-Cabernet', 'Sauvign

In [31]:
avg = []
for i in grapes:
    avg.append(df_wine[df_wine.variety == i]['points'].mean())

##### 3.11 Get the best wines overall based on the wine tasters' ratings

In [32]:
wine = df_wine['title'].unique()
wine

array(['Nicosia 2013 Vulkà Bianco  (Etna)',
       'Quinta dos Avidagos 2011 Avidagos Red (Douro)',
       'Rainstorm 2013 Pinot Gris (Willamette Valley)', ...,
       'Domaine Gresser 2013 Kritt Gewurztraminer (Alsace)',
       'Domaine Marcel Deiss 2012 Pinot Gris (Alsace)',
       'Domaine Schoffit 2012 Lieu-dit Harth Cuvée Caroline Gewurztraminer (Alsace)'],
      dtype=object)

In [ ]:
avg = []
for i in wine:
    avg.append(df_wine[df_wine.title == i]['points'].mean())

In [ ]:
wine_avg_points = {'Wine':wine, 'Average Points': avg} 

wine_avg_points = pd.DataFrame(wine_avg_points) 
 
wine_avg_points = wine_avg_points.sort_values(by=['Average Points'], ascending=False)

wine_avg_points[wine_avg_points['Average Points'] == 100]

##### 3.12 Find out if age (in year) determines the the wine quality (represented by its score)

In [ ]:
age = []
age = df_wine['age'].value_counts().keys().to_list()
average_age = []
for a in age:
    average_age.append(df_wine[df_wine['age'] == a].points.mean())
df_age = pd.DataFrame({'age':age, 'points':average_age})
df_age = df_age.sort_values(by=['age'], ascending=True).reset_index(drop=True)
df_age = df_age.set_index('age')

In [ ]:
df_age

In [ ]:
df_age.plot(kind='line', figsize=(40,20))

Analyzing the data comparing the points of the wines based on their age. Even though there are some extreme data and outliers, such as the significant decrease in points for wines of 40 years old, we can still see the gradual increase in points for wines from 0 years-old to around 50 years-old. After that, the points start to decrease as the wines’ age increases. The graph shows a distribution similar to a Weibull distribution.

##### 3.13 Find out if price (in USD) determines the the wine quality (represented by its score)

In [ ]:
price = []
price = df_wine['price'].value_counts().keys().to_list()
average_price = []
for a in price:
    average_price.append(df_wine[df_wine['price'] == a].points.mean())
df_price = pd.DataFrame({'price':price, 'points':average_price})
df_price = df_price.sort_values(by=['price'], ascending=True).reset_index(drop=True)
df_price = df_price.set_index('price')

In [ ]:
df_price

In [ ]:
df_price.plot(kind='line', figsize=(40,20))

Analyzing the data comparing the points of the wines based on their price. There is no very clear distribution of this data. In other words, with the same price, there can be both very high or very low points given. It would be very difficult (or unlikely) to see the pattern of points by looking only at the price.

##### 3.14 Give suggestion to each wine taster based on their preferred winery and variety

In [ ]:
def getSuggestion(dfWinery, dfVariety, tasterName):
    return df_wine.loc[((df_wine['winery'] == dfWinery.index.tolist()[0])
                                  | (df_wine['winery'] == dfWinery.index.tolist()[1])
                                  | (df_wine['winery'] == dfWinery.index.tolist()[2])
                                  | (df_wine['winery'] == dfWinery.index.tolist()[3])
                                  | (df_wine['winery'] == dfWinery.index.tolist()[4])
                                  | (df_wine['winery'] == dfWinery.index.tolist()[0])
                                  | (df_wine['variety'] == dfVariety.index.tolist()[1])
                                  | (df_wine['variety'] == dfVariety.index.tolist()[2])
                                  | (df_wine['variety'] == dfVariety.index.tolist()[3])
                                  | (df_wine['variety'] == dfVariety.index.tolist()[4]))
                                 & (df_wine['taster_name'] != tasterName)]

Wine suggestion list for Roger Voss

In [ ]:
df_voss = df_wine.loc[df_wine['taster_name'] == 'Roger Voss']
df_voss = df_voss.reset_index(drop=True)

In [ ]:
winery_voss = []
winery_voss = df_voss['winery'].value_counts().keys().to_list()
average_winery_voss = []
for a in winery_voss:
    average_winery_voss.append(df_voss[df_voss['winery'] == a].points.mean())
df_winery_voss = pd.DataFrame({'winery':winery_voss, 'points':average_winery_voss})
df_winery_voss = df_winery_voss.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_voss = df_winery_voss.set_index('winery')

In [ ]:
variety_voss = []
variety_voss = df_voss['variety'].value_counts().keys().to_list()
average_variety_voss = []
for a in variety_voss:
    average_variety_voss.append(df_voss[df_voss['variety'] == a].points.mean())
df_variety_voss = pd.DataFrame({'variety':variety_voss, 'points':average_variety_voss})
df_variety_voss = df_variety_voss.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_voss = df_variety_voss.set_index('variety')

In [ ]:
df_suggestion_voss = getSuggestion(df_winery_voss, df_variety_voss, 'Roger Voss')
df_suggestion_voss = df_suggestion_voss.sort_values(by=['points'], ascending=False).head()
df_suggestion_voss

Wine suggestion list for Michael Schachner

In [ ]:
df_schachner = df_wine.loc[df_wine['taster_name'] == 'Michael Schachner']
df_schachner = df_schachner.reset_index(drop=True)

In [ ]:
winery_schachner = []
winery_schachner = df_schachner['winery'].value_counts().keys().to_list()
average_winery_schachner = []
for a in winery_schachner:
    average_winery_schachner.append(df_schachner[df_schachner['winery'] == a].points.mean())
df_winery_schachner = pd.DataFrame({'winery':winery_schachner, 'points':average_winery_schachner})
df_winery_schachner = df_winery_schachner.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_schachner = df_winery_schachner.set_index('winery')

In [ ]:
variety_schachner = []
variety_schachner = df_schachner['variety'].value_counts().keys().to_list()
average_variety_schachner = []
for a in variety_schachner:
    average_variety_schachner.append(df_schachner[df_schachner['variety'] == a].points.mean())
df_variety_schachner = pd.DataFrame({'variety':variety_schachner, 'points':average_variety_schachner})
df_variety_schachner = df_variety_schachner.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_schachner = df_variety_schachner.set_index('variety')

In [ ]:
df_suggestion_schachner = getSuggestion(df_winery_schachner, df_variety_schachner, 'Michael Schachner')
df_suggestion_schachner = df_suggestion_schachner.sort_values(by=['points'], ascending=False).head()
df_suggestion_schachner

Wine suggestion list for Kerin O’Keefe

In [ ]:
df_keefe = df_wine.loc[df_wine['taster_name'] == 'Kerin O’Keefe']
df_keefe = df_keefe.reset_index(drop=True)

In [ ]:
winery_keefe = []
winery_keefe = df_keefe['winery'].value_counts().keys().to_list()
average_winery_keefe = []
for a in winery_keefe:
    average_winery_keefe.append(df_keefe[df_keefe['winery'] == a].points.mean())
df_winery_keefe = pd.DataFrame({'winery':winery_keefe, 'points':average_winery_keefe})
df_winery_keefe = df_winery_keefe.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_keefe = df_winery_keefe.set_index('winery')

In [ ]:
variety_keefe = []
variety_keefe = df_keefe['variety'].value_counts().keys().to_list()
average_variety_keefe = []
for a in variety_keefe:
    average_variety_keefe.append(df_keefe[df_keefe['variety'] == a].points.mean())
df_variety_keefe = pd.DataFrame({'variety':variety_keefe, 'points':average_variety_keefe})
df_variety_keefe = df_variety_keefe.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_keefe = df_variety_keefe.set_index('variety')

In [ ]:
df_suggestion_keefe = getSuggestion(df_winery_keefe, df_variety_keefe, 'Kerin O’Keefe')
df_suggestion_keefe = df_suggestion_keefe.sort_values(by=['points'], ascending=False).head()
df_suggestion_keefe

Wine suggestion list for Virginie Boone

In [ ]:
df_boone = df_wine.loc[df_wine['taster_name'] == 'Virginie Boone']
df_boone = df_boone.reset_index(drop=True)

In [ ]:
winery_boone = []
winery_boone = df_boone['winery'].value_counts().keys().to_list()
average_winery_boone = []
for a in winery_boone:
    average_winery_boone.append(df_boone[df_boone['winery'] == a].points.mean())
df_winery_boone = pd.DataFrame({'winery':winery_boone, 'points':average_winery_boone})
df_winery_boone = df_winery_boone.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_boone = df_winery_boone.set_index('winery')

In [ ]:
variety_boone = []
variety_boone = df_boone['variety'].value_counts().keys().to_list()
average_variety_boone = []
for a in variety_boone:
    average_variety_boone.append(df_boone[df_boone['variety'] == a].points.mean())
df_variety_boone = pd.DataFrame({'variety':variety_boone, 'points':average_variety_boone})
df_variety_boone = df_variety_boone.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_boone = df_variety_boone.set_index('variety')

In [ ]:
df_suggestion_boone = getSuggestion(df_winery_boone, df_variety_boone, 'Virginie Boone')
df_suggestion_boone = df_suggestion_boone.sort_values(by=['points'], ascending=False).head()
df_suggestion_boone

Wine suggestion list for Paul Gregutt

In [ ]:
df_gregutt = df_wine.loc[df_wine['taster_name'] == 'Paul Gregutt']
df_gregutt = df_gregutt.reset_index(drop=True)

In [ ]:
winery_gregutt = []
winery_gregutt = df_gregutt['winery'].value_counts().keys().to_list()
average_winery_gregutt = []
for a in winery_gregutt:
    average_winery_gregutt.append(df_gregutt[df_gregutt['winery'] == a].points.mean())
df_winery_gregutt = pd.DataFrame({'winery':winery_gregutt, 'points':average_winery_gregutt})
df_winery_gregutt = df_winery_gregutt.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_gregutt = df_winery_gregutt.set_index('winery')

In [ ]:
variety_gregutt = []
variety_gregutt = df_gregutt['variety'].value_counts().keys().to_list()
average_variety_gregutt = []
for a in variety_gregutt:
    average_variety_gregutt.append(df_gregutt[df_gregutt['variety'] == a].points.mean())
df_variety_gregutt = pd.DataFrame({'variety':variety_gregutt, 'points':average_variety_gregutt})
df_variety_gregutt = df_variety_gregutt.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_gregutt = df_variety_gregutt.set_index('variety')

In [ ]:
df_suggestion_gregutt = getSuggestion(df_winery_gregutt, df_variety_gregutt, 'Virginie Boone')
df_suggestion_gregutt = df_suggestion_gregutt.sort_values(by=['points'], ascending=False).head()
df_suggestion_gregutt

Wine suggestion list for Matt Kettmann

In [ ]:
df_kettmann = df_wine.loc[df_wine['taster_name'] == 'Matt Kettmann']
df_kettmann = df_kettmann.reset_index(drop=True)

In [ ]:
winery_kettmann = []
winery_kettmann = df_kettmann['winery'].value_counts().keys().to_list()
average_winery_kettmann = []
for a in winery_kettmann:
    average_winery_kettmann.append(df_kettmann[df_kettmann['winery'] == a].points.mean())
df_winery_kettmann = pd.DataFrame({'winery':winery_kettmann, 'points':average_winery_kettmann})
df_winery_kettmann = df_winery_kettmann.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_kettmann = df_winery_kettmann.set_index('winery')

In [ ]:
variety_kettmann = []
variety_kettmann = df_kettmann['variety'].value_counts().keys().to_list()
average_variety_kettmann = []
for a in variety_kettmann:
    average_variety_kettmann.append(df_kettmann[df_kettmann['variety'] == a].points.mean())
df_variety_kettmann = pd.DataFrame({'variety':variety_kettmann, 'points':average_variety_kettmann})
df_variety_kettmann = df_variety_kettmann.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_kettmann = df_variety_kettmann.set_index('variety')

In [ ]:
df_suggestion_kettmann = getSuggestion(df_winery_kettmann, df_variety_kettmann, 'Matt Kettmann')
df_suggestion_kettmann = df_suggestion_kettmann.sort_values(by=['points'], ascending=False).head()
df_suggestion_kettmann

Wine suggestion list for Joe Czerwinski

In [ ]:
df_czerwinski = df_wine.loc[df_wine['taster_name'] == 'Joe Czerwinski']
df_czerwinski = df_czerwinski.reset_index(drop=True)

In [ ]:
winery_czerwinski = []
winery_czerwinski = df_czerwinski['winery'].value_counts().keys().to_list()
average_winery_czerwinski = []
for a in winery_czerwinski:
    average_winery_czerwinski.append(df_czerwinski[df_czerwinski['winery'] == a].points.mean())
df_winery_czerwinski = pd.DataFrame({'winery':winery_czerwinski, 'points':average_winery_czerwinski})
df_winery_czerwinski = df_winery_czerwinski.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_czerwinski = df_winery_czerwinski.set_index('winery')

In [ ]:
variety_czerwinski = []
variety_czerwinski = df_czerwinski['variety'].value_counts().keys().to_list()
average_variety_czerwinski = []
for a in variety_czerwinski:
    average_variety_czerwinski.append(df_czerwinski[df_czerwinski['variety'] == a].points.mean())
df_variety_czerwinski = pd.DataFrame({'variety':variety_czerwinski, 'points':average_variety_czerwinski})
df_variety_czerwinski = df_variety_czerwinski.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_czerwinski = df_variety_czerwinski.set_index('variety')

In [ ]:
df_suggestion_czerwinski = getSuggestion(df_winery_czerwinski, df_variety_czerwinski, 'Joe Czerwinski')
df_suggestion_czerwinski = df_suggestion_czerwinski.sort_values(by=['points'], ascending=False).head()
df_suggestion_czerwinski

Wine suggestion list for Sean P. Sullivan

In [ ]:
df_sullivan = df_wine.loc[df_wine['taster_name'] == 'Sean P. Sullivan']
df_sullivan = df_sullivan.reset_index(drop=True)

In [ ]:
winery_sullivan = []
winery_sullivan = df_sullivan['winery'].value_counts().keys().to_list()
average_winery_sullivan = []
for a in winery_sullivan:
    average_winery_sullivan.append(df_sullivan[df_sullivan['winery'] == a].points.mean())
df_winery_sullivan = pd.DataFrame({'winery':winery_sullivan, 'points':average_winery_sullivan})
df_winery_sullivan = df_winery_sullivan.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_sullivan = df_winery_sullivan.set_index('winery')

In [ ]:
variety_sullivan = []
variety_sullivan = df_sullivan['variety'].value_counts().keys().to_list()
average_variety_sullivan = []
for a in variety_sullivan:
    average_variety_sullivan.append(df_sullivan[df_sullivan['variety'] == a].points.mean())
df_variety_sullivan = pd.DataFrame({'variety':variety_sullivan, 'points':average_variety_sullivan})
df_variety_sullivan = df_variety_sullivan.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_sullivan = df_variety_sullivan.set_index('variety')

In [ ]:
df_suggestion_sullivan = getSuggestion(df_winery_sullivan, df_variety_sullivan, 'Sean P. Sullivan')
df_suggestion_sullivan = df_suggestion_sullivan.sort_values(by=['points'], ascending=False).head()
df_suggestion_sullivan

Wine suggestion list for Anna Lee C. Iijima

In [ ]:
df_iijima = df_wine.loc[df_wine['taster_name'] == 'Anna Lee C. Iijima']
df_iijima = df_iijima.reset_index(drop=True)

In [ ]:
winery_iijima = []
winery_iijima = df_iijima['winery'].value_counts().keys().to_list()
average_winery_iijima = []
for a in winery_iijima:
    average_winery_iijima.append(df_iijima[df_iijima['winery'] == a].points.mean())
df_winery_iijima = pd.DataFrame({'winery':winery_iijima, 'points':average_winery_iijima})
df_winery_iijima = df_winery_iijima.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_iijima = df_winery_iijima.set_index('winery')

In [ ]:
variety_iijima = []
variety_iijima = df_iijima['variety'].value_counts().keys().to_list()
average_variety_iijima = []
for a in variety_iijima:
    average_variety_iijima.append(df_iijima[df_iijima['variety'] == a].points.mean())
df_variety_iijima = pd.DataFrame({'variety':variety_iijima, 'points':average_variety_iijima})
df_variety_iijima = df_variety_iijima.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_iijima = df_variety_iijima.set_index('variety')

In [ ]:
df_suggestion_iijima = getSuggestion(df_winery_iijima, df_variety_iijima, 'Anna Lee C. Iijima')
df_suggestion_iijima = df_suggestion_iijima.sort_values(by=['points'], ascending=False).head()
df_suggestion_iijima

Wine suggestion list for Jim Gordon

In [ ]:
df_gordon = df_wine.loc[df_wine['taster_name'] == 'Jim Gordon']
df_gordon = df_gordon.reset_index(drop=True)

In [ ]:
winery_gordon = []
winery_gordon = df_gordon['winery'].value_counts().keys().to_list()
average_winery_gordon = []
for a in winery_gordon:
    average_winery_gordon.append(df_gordon[df_gordon['winery'] == a].points.mean())
df_winery_gordon = pd.DataFrame({'winery':winery_gordon, 'points':average_winery_gordon})
df_winery_gordon = df_winery_gordon.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_gordon = df_winery_gordon.set_index('winery')

In [ ]:
variety_gordon = []
variety_gordon = df_gordon['variety'].value_counts().keys().to_list()
average_variety_gordon = []
for a in variety_gordon:
    average_variety_gordon.append(df_gordon[df_gordon['variety'] == a].points.mean())
df_variety_gordon = pd.DataFrame({'variety':variety_gordon, 'points':average_variety_gordon})
df_variety_gordon = df_variety_gordon.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_gordon = df_variety_gordon.set_index('variety')

In [ ]:
df_suggestion_gordon = getSuggestion(df_winery_gordon, df_variety_gordon, 'Jim Gordon')
df_suggestion_gordon = df_suggestion_gordon.sort_values(by=['points'], ascending=False).head()
df_suggestion_gordon

Wine suggestion list for Lauren Buzzeo

In [ ]:
df_buzzeo = df_wine.loc[df_wine['taster_name'] == 'Lauren Buzzeo']
df_buzzeo = df_buzzeo.reset_index(drop=True)

In [ ]:
winery_buzzeo = []
winery_buzzeo = df_buzzeo['winery'].value_counts().keys().to_list()
average_winery_buzzeo = []
for a in winery_buzzeo:
    average_winery_buzzeo.append(df_buzzeo[df_buzzeo['winery'] == a].points.mean())
df_winery_buzzeo = pd.DataFrame({'winery':winery_buzzeo, 'points':average_winery_buzzeo})
df_winery_buzzeo = df_winery_buzzeo.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_buzzeo = df_winery_buzzeo.set_index('winery')

In [ ]:
variety_buzzeo = []
variety_buzzeo = df_buzzeo['variety'].value_counts().keys().to_list()
average_variety_buzzeo = []
for a in variety_buzzeo:
    average_variety_buzzeo.append(df_buzzeo[df_buzzeo['variety'] == a].points.mean())
df_variety_buzzeo = pd.DataFrame({'variety':variety_buzzeo, 'points':average_variety_buzzeo})
df_variety_buzzeo = df_variety_buzzeo.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_buzzeo = df_variety_buzzeo.set_index('variety')

In [ ]:
df_suggestion_buzzeo = getSuggestion(df_winery_buzzeo, df_variety_buzzeo, 'Lauren Buzzeo')
df_suggestion_buzzeo = df_suggestion_buzzeo.sort_values(by=['points'], ascending=False).head()
df_suggestion_buzzeo

Wine suggestion list for Susan Kostrzewa

In [ ]:
df_kostrzewa = df_wine.loc[df_wine['taster_name'] == 'Susan Kostrzewa']
df_kostrzewa = df_kostrzewa.reset_index(drop=True)

In [ ]:
winery_kostrzewa = []
winery_kostrzewa = df_kostrzewa['winery'].value_counts().keys().to_list()
average_winery_kostrzewa = []
for a in winery_kostrzewa:
    average_winery_kostrzewa.append(df_kostrzewa[df_kostrzewa['winery'] == a].points.mean())
df_winery_kostrzewa = pd.DataFrame({'winery':winery_kostrzewa, 'points':average_winery_kostrzewa})
df_winery_kostrzewa = df_winery_kostrzewa.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_kostrzewa = df_winery_kostrzewa.set_index('winery')

In [ ]:
variety_kostrzewa = []
variety_kostrzewa = df_kostrzewa['variety'].value_counts().keys().to_list()
average_variety_kostrzewa = []
for a in variety_kostrzewa:
    average_variety_kostrzewa.append(df_kostrzewa[df_kostrzewa['variety'] == a].points.mean())
df_variety_kostrzewa = pd.DataFrame({'variety':variety_kostrzewa, 'points':average_variety_kostrzewa})
df_variety_kostrzewa = df_variety_kostrzewa.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_kostrzewa = df_variety_kostrzewa.set_index('variety')

In [ ]:
df_suggestion_kostrzewa = getSuggestion(df_winery_kostrzewa, df_variety_kostrzewa, 'Susan Kostrzewa')
df_suggestion_kostrzewa = df_suggestion_kostrzewa.sort_values(by=['points'], ascending=False).head()
df_suggestion_kostrzewa

Wine suggestion list for Mike DeSimone

In [ ]:
df_desimone = df_wine.loc[df_wine['taster_name'] == 'Mike DeSimone']
df_desimone = df_desimone.reset_index(drop=True)

In [ ]:
winery_desimone = []
winery_desimone = df_desimone['winery'].value_counts().keys().to_list()
average_winery_desimone = []
for a in winery_desimone:
    average_winery_desimone.append(df_desimone[df_desimone['winery'] == a].points.mean())
df_winery_desimone = pd.DataFrame({'winery':winery_desimone, 'points':average_winery_desimone})
df_winery_desimone = df_winery_desimone.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_desimone = df_winery_desimone.set_index('winery')

In [ ]:
variety_desimone = []
variety_desimone = df_desimone['variety'].value_counts().keys().to_list()
average_variety_desimone = []
for a in variety_desimone:
    average_variety_desimone.append(df_desimone[df_desimone['variety'] == a].points.mean())
df_variety_desimone = pd.DataFrame({'variety':variety_desimone, 'points':average_variety_desimone})
df_variety_desimone = df_variety_desimone.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_desimone = df_variety_desimone.set_index('variety')

In [ ]:
df_suggestion_desimone = getSuggestion(df_winery_desimone, df_variety_desimone, 'Mike DeSimone')
df_suggestion_desimone = df_suggestion_desimone.sort_values(by=['points'], ascending=False).head()
df_suggestion_desimone

Wine suggestion list for Jeff Jenssen

In [ ]:
df_jenssen = df_wine.loc[df_wine['taster_name'] == 'Jeff Jenssen']
df_jenssen = df_jenssen.reset_index(drop=True)

In [ ]:
winery_jenssen = []
winery_jenssen = df_jenssen['winery'].value_counts().keys().to_list()
average_winery_jenssen = []
for a in winery_jenssen:
    average_winery_jenssen.append(df_jenssen[df_jenssen['winery'] == a].points.mean())
df_winery_jenssen = pd.DataFrame({'winery':winery_jenssen, 'points':average_winery_jenssen})
df_winery_jenssen = df_winery_jenssen.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_jenssen = df_winery_jenssen.set_index('winery')

variety_jenssen = []
variety_jenssen = df_jenssen['variety'].value_counts().keys().to_list()
average_variety_jenssen = []
for a in variety_jenssen:
    average_variety_jenssen.append(df_jenssen[df_jenssen['variety'] == a].points.mean())
df_variety_jenssen = pd.DataFrame({'variety':variety_jenssen, 'points':average_variety_jenssen})
df_variety_jenssen = df_variety_jenssen.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_jenssen = df_variety_jenssen.set_index('variety')

df_suggestion_jenssen = getSuggestion(df_winery_jenssen, df_variety_jenssen, 'Jeff Jenssen')
df_suggestion_jenssen = df_suggestion_jenssen.sort_values(by=['points'], ascending=False).head()
df_suggestion_jenssen

Wine suggestion list for Alexander Peartree

In [ ]:
df_peartree = df_wine.loc[df_wine['taster_name'] == 'Alexander Peartree']
df_peartree = df_peartree.reset_index(drop=True)

In [ ]:
winery_peartree = []
winery_peartree = df_peartree['winery'].value_counts().keys().to_list()
average_winery_peartree = []
for a in winery_peartree:
    average_winery_peartree.append(df_peartree[df_peartree['winery'] == a].points.mean())
df_winery_peartree = pd.DataFrame({'winery':winery_peartree, 'points':average_winery_peartree})
df_winery_peartree = df_winery_peartree.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_peartree = df_winery_peartree.set_index('winery')

In [ ]:
variety_peartree = []
variety_peartree = df_peartree['variety'].value_counts().keys().to_list()
average_variety_peartree = []
for a in variety_peartree:
    average_variety_peartree.append(df_peartree[df_peartree['variety'] == a].points.mean())
df_variety_peartree = pd.DataFrame({'variety':variety_peartree, 'points':average_variety_peartree})
df_variety_peartree = df_variety_peartree.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_peartree = df_variety_peartree.set_index('variety')

In [ ]:
df_suggestion_peartree = getSuggestion(df_winery_peartree, df_variety_peartree, 'Alexander Peartree')
df_suggestion_peartree = df_suggestion_peartree.sort_values(by=['points'], ascending=False).head()
df_suggestion_peartree

Wine suggestion list for Carrie Dykes

In [ ]:
df_dykes = df_wine.loc[df_wine['taster_name'] == 'Carrie Dykes']
df_dykes = df_dykes.reset_index(drop=True)

In [ ]:
winery_dykes = []
winery_dykes = df_dykes['winery'].value_counts().keys().to_list()
average_winery_dykes = []
for a in winery_dykes:
    average_winery_dykes.append(df_dykes[df_dykes['winery'] == a].points.mean())
df_winery_dykes = pd.DataFrame({'winery':winery_dykes, 'points':average_winery_dykes})
df_winery_dykes = df_winery_dykes.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_dykes = df_winery_dykes.set_index('winery')

In [ ]:
variety_dykes = []
variety_dykes = df_dykes['variety'].value_counts().keys().to_list()
average_variety_dykes = []
for a in variety_dykes:
    average_variety_dykes.append(df_dykes[df_dykes['variety'] == a].points.mean())
df_variety_dykes = pd.DataFrame({'variety':variety_dykes, 'points':average_variety_dykes})
df_variety_dykes = df_variety_dykes.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_dykes = df_variety_dykes.set_index('variety')

In [ ]:
df_suggestion_dykes = getSuggestion(df_winery_dykes, df_variety_dykes, 'Carrie Dykes')
df_suggestion_dykes = df_suggestion_dykes.sort_values(by=['points'], ascending=False).head()
df_suggestion_dykes

Wine suggestion list for Fiona Adams

In [ ]:
df_adams = df_wine.loc[df_wine['taster_name'] == 'Fiona Adams']
df_adams = df_adams.reset_index(drop=True)

In [ ]:
winery_adams = []
winery_adams = df_adams['winery'].value_counts().keys().to_list()
average_winery_adams = []
for a in winery_adams:
    average_winery_adams.append(df_adams[df_adams['winery'] == a].points.mean())
df_winery_adams = pd.DataFrame({'winery':winery_adams, 'points':average_winery_adams})
df_winery_adams = df_winery_adams.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_adams = df_winery_adams.set_index('winery')

In [ ]:
variety_adams = []
variety_adams = df_adams['variety'].value_counts().keys().to_list()
average_variety_adams = []
for a in variety_adams:
    average_variety_adams.append(df_adams[df_adams['variety'] == a].points.mean())
df_variety_adams = pd.DataFrame({'variety':variety_adams, 'points':average_variety_adams})
df_variety_adams = df_variety_adams.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_adams = df_variety_adams.set_index('variety')

In [ ]:
df_suggestion_adams = getSuggestion(df_winery_adams, df_variety_adams, 'Fiona Adams')
df_suggestion_adams = df_suggestion_adams.sort_values(by=['points'], ascending=False).head()
df_suggestion_adams

Wine suggestion list for Christina Pickard

In [ ]:
df_pickard = df_wine.loc[df_wine['taster_name'] == 'Christina Pickard']
df_pickard = df_pickard.reset_index(drop=True)

In [ ]:
winery_pickard = []
winery_pickard = df_pickard['winery'].value_counts().keys().to_list()
average_winery_pickard = []
for a in winery_pickard:
    average_winery_pickard.append(df_pickard[df_pickard['winery'] == a].points.mean())
df_winery_pickard = pd.DataFrame({'winery':winery_pickard, 'points':average_winery_pickard})
df_winery_pickard = df_winery_pickard.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_winery_pickard = df_winery_pickard.set_index('winery')

In [ ]:
variety_pickard = []
variety_pickard = df_pickard['variety'].value_counts().keys().to_list()
average_variety_pickard = []
for a in variety_pickard:
    average_variety_pickard.append(df_pickard[df_pickard['variety'] == a].points.mean())
df_variety_pickard = pd.DataFrame({'variety':variety_pickard, 'points':average_variety_pickard})
df_variety_pickard = df_variety_pickard.sort_values(by=['points'], ascending=False).reset_index(drop=True)
df_variety_pickard = df_variety_pickard.set_index('variety')

In [ ]:
df_suggestion_pickard = getSuggestion(df_winery_pickard, df_variety_pickard, 'Christina Pickard')
df_suggestion_pickard = df_suggestion_pickard.sort_values(by=['points'], ascending=False).head()
df_suggestion_pickard

While we need to compare the wines based on where they are made and what kind of grapes they were made out of, we group the rows that have the same value in specific columns (i.e. country, winery) and take the mean of the point values. For comparing wines from different continents, we create a function that will group the countries based on the continents they’re in, then do the same thing for calculating each continent’s point. We took the mean of the points because we should be looking at the average/mean of the points to get a fair analysis of the data.

In order to answer question 4 and 6, we have to see how rating points as the age or price of the wine increases. Thus, we created a function that will calculate the age of the wine and created two line plots, one for price and one for age, to visualize the price change along with age and price. We chose to use line plot in order to see how points change based on age/price.

For the last part of the analysis, we want to know which wine would each wine taster recommend. Hence, we group the data based on the tasters’ names and sort them based on the wines’ points. The highest pointed wine would likely be the wine each taster would recommend.

# Ethics & Privacy

The data we used is free for public use and can be collected from a website of wine reviews. It provides its licence on CC BY-NC-SA 4.0 that says we are free to share (copy and redistribute the material in any medium or format) and adapt (remix, transform, and build upon the material). As we go through our datasets, any data containing personal information will be discarded unhesitatingly to secure the privacy of the wine testers, for example their twitter accounts. Therefore, there will be no personal data that will be used inappropriately. There are potential sources of bias with this data. This dataset only contains data from WineEnthusiast. If there are other relevant big datasets similar to what we use, that may be problematic in terms of it allowing for equitable analysis. There can possibly be different answers within either of the two big datasets, if there are any, causing bias in our analysis. Adding on to that, one of our dataset variables lists out nineteen wine testers who gave out all the wine ratings. This is also likely to reflect human biases in a way that could be a problem, however, the only rational way to rate the taste quality of wines is to actually try it on humans. There is as well another bias as in the dataset as it compiles the scores of wines that are only 80 points above which gives us some incomplete findings unlike if we were to use all wine ratings from 0 to 100 points.

# Conclusion & Discussion

According to our hypothesis of the best wine, we thought it would be the Chateau Montelena Napa Valley Chardonnay 2016 and the grape type of Chardonnay. We were wrong in our initial hypothesis because according to our findings, the best wine overall in terms of ratings are Chateau, Sauvignon, etc. There are 17 wines that might be good recommendations  to the public consumers as they were best picked by 19 professional wine testers (sommeliers) with perfect score of 100. These wines include: Salon 2006 Le Mesnil Blanc de Blancs Brut, Château Léoville Las Cases 2010 Saint-Julien, Casa Ferreirinha 2008 Barca-Velha Red (Douro), Château Léoville Barton 2010 Saint-Julien, Cayuse 2008 Bionic Frog Syrah, and 12 other wines. A type of grape is a big factor that determines the taste of the wine since they are what wine is all about after all. We were wrong about our hypothesis that the best grapes being Chardonnay. From our data analysis, we were able to get the type of grape that produces the best wine with a common type of grape is Tinda del Pais, giving us a rating of 95 out of 100.

We also missed a bit from our hypothesis on analyzing the best wine in terms of continent and country. We hypothesize that Europe would produce the most high-rated wines among other continents which we were right about. Europe has the average of 88.609 out of 100, slightly higher than North America with 0.051609%. However, in terms of country, our initial thought that France would be the top wine country was wrong. Based on our data analysis, we could obviously see that England is the country that produces the highest-rated wines with the average point of 91.581 out of 100, higher than the second leading country, India, with 1.506%. However, a slight contrary to that result occured as we were getting more information from our dataset. We collected that the best winery with the best ratings of wine overall is in Araujo with average point of 98 out of 100, and the second one is Gandona with 97 out of 100, although both of them are located in California.

For our next hypothesis, we thought as the wine ages more, they would increase in quality reflected in their price value, which also directly means that they would increase in terms of their ratings as wine enthusiasts would then enjoy aged wine more. However, we were only partially correct as it turned out that the age mattered but not the price. According to our graph analysis, the wine ratings given gradually increases as the wine ages whereas the price fluctuates so much that we could not come to a conclusion that price contributes to the ratings.

We wanted to further our analysis and we then wanted to use our report to include the questions of which wines would be a good recommendation to each wine tasters based on their preferred winery and types of grapes. We were able to get the wine preference of each of the tasters and we are able to give them suggestions based on the wines they have evaluated. For example, one of the wine taster, Roger Voss, we successfully listed out 5 top wine suggestions based on their high-scored winery and type of grapes: Salon 1985 Le Mesnil Blanc de Blancs Brut, Salon 1988 Le Mesnil Blanc de Blancs Brut, Knipser 2008 Trocken Chardonnay Weissburgunder, Krug 1988 Brut (Champagne),  and Krug NV Grande Cuvée Brut (Champagne).

On top of all our analysis, we would like to predict wine ratings that would be given to a new wine never tested before based on its taster name, winery, variety, country, continent, age and price if they have correlation.


# Team Contributions

We first brainstormed on what kind of data we would like to work on together and came up to the conclusion of the wine testing. Natalia Sudjana and Gabriela Shirley first did background research on the wine testing. Devin Junanto as well as Shih Gau Peng then proceeded to do the data wrangling as they cleaned up the data sets and took whatever is necessary for our analysis while Natalia and Gabriela were refining the questions, hypothesis as well as doing the reports. Natalia and Shirley then started to do the analysis after the clean up and drew on correlations and linked necessary datas to answer the research questions. Devin and Shih Gau then made the necessary adjustments as edits to clean up our data. We discussed as a team of our findings to speak of our final thoughts and finalize the results. Natalia and Shirley then recorded them to the final report.